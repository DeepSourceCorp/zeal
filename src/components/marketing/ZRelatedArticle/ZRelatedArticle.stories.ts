import ZRelatedArticle from './ZRelatedArticle.vue'

export default {
  title: 'Zeal Marketing/Related Article',
  component: ZRelatedArticle,
  excludeStories: /.*Data$/
}

export const Basic = () => ({
  components: { ZRelatedArticle },
  template: `<div class='padded-container bg-ink-400 text-vanilla-400 grid grid-cols-12'>
      <z-related-article class="col-span-8">
        <span slot="heading">Odio amet tincidunt metus dictum dapibus cum nibh sed</span>
        <a href="/somethign" slot="link">Read More</a>
      </z-related-article>
    </div>`
})