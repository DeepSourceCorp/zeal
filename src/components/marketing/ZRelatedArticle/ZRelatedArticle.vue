<template>
  <span class="grid grid-cols-12 py-3">
    <span class="col-span-10 mb-2 block text-vanilla-200 text-lg font-medium leading-9">
      <slot name="heading"></slot>
    </span>
    <span class="flex items-center col-span-10 text-sm font-medium uppercase text-juniper">
      <slot name="link"></slot>
      <z-icon color="juniper" class="ml-1" size="small" icon="arrow-right"></z-icon>
    </span>
  </span>
</template>

<script lang="ts">
import Vue from 'vue'
import ZIcon from '../../ZIcon'

export default Vue.extend({
  name: 'ZRelatedArticle',
  components: {
    ZIcon
  }
})
</script>
