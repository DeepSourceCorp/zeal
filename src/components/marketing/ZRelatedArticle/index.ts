import ZRelatedArticle from './ZRelatedArticle.vue'

export default ZRelatedArticle
