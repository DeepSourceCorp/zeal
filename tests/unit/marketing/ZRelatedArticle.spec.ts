import Vue from 'vue'
import { createLocalVue, mount, Wrapper } from '@vue/test-utils'

import ZRelatedArticle from '../../../src/components/marketing/ZRelatedArticle'

describe('Related Article component', () => {
  let mountFunction: (options?: Record<string, unknown>) => Wrapper<Vue>
  let localVue: typeof Vue
  beforeEach(() => {
    localVue = createLocalVue()
    mountFunction = (options = {}) => {
      return mount(ZRelatedArticle, {
        localVue,
        ...options
      })
    }
  })
  it('renders a related article when the component is mounted', () => {
    expect(mountFunction().html()).toMatchSnapshot()
  })
})
