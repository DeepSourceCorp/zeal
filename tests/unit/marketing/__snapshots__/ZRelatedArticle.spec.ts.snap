// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Related Article component renders a related article when the component is mounted 1`] = `<span class="grid grid-cols-12 py-3"><span class="col-span-10 mb-2 block text-lg font-medium leading-9"></span> <span class="flex items-center col-span-10 text-sm font-medium uppercase text-juniper"> <svg viewBox="0 0 24 24" fill="none" class="stroke-2 ml-1 z-icon w-4 h-4 z-icon--arrow-right text-juniper"><line x1="5" y1="12" x2="19" y2="12"></line><polyline points="12 5 19 12 12 19"></polyline></svg></span></span>`;
